src/main/java/animals/Sparrow.java

```diff
@@ -0,0 +1,31 @@
+package animals;
+
+import areas.BirdAviary;
+import interfaces.Flyable;
+
+public class Sparrow extends Animal implements Flyable, BirdAviary {
+    private String name;
```

Comment:
this field is repeated in several classes, that's why it's better to move to your abstract class, so we will avoid code duplication

Reply:
I don't understand this
I added declaration of private String name in abstract class animal but I cant delete this declaration of name in classes like Penguin, Bird, Shark and Tiger.
Can you explain it please?

Reply:
understood it after task with figures
pushed with fixes, check please